### PR TITLE
New Template - cisco_asa show_module

### PR DESCRIPTION
**Pull request by lamiskin**
##### ISSUE TYPE

 - New Template Pull Request

##### COMPONENT

ntc_templates/templates/cisco_asa_show_module.textfsm
cisco_asa
show module

##### SUMMARY

Some ASAs support `show module` output. Not all in my experience.
Sub templates were used as they are with cisco_ios show_module.
No data is collected on the 3rd section, as I don't have any example data to base it off.

```
parsed_sample:
  - module: "0"
    cardtype: "ASA 5585-X Security Services Processor-40 w"
    model: "ASA5585-SSP-40"
    serial: "JAF499EDRD"
    mac_addr: "30e4.db11.1111 to 30e4.db22.2222"
    mod_hw: "1.1"
    mod_fw: "2.0(7)2"
    mod_sw: "8.4(2)"
    status: "Up Sys"

```

### CODE REVIEW

**Review by diepes — APPROVED** [review]

File: ntc_templates/templates/index [hunk 1]
```diff
@@ -150,6 +150,7 @@ ciena_saos_port_show.textfsm, .*, ciena_saos, po[[rt]] sh[[ow]]
 ciena_saos_rstp_show.textfsm, .*, ciena_saos, rs[[tp]] sh[[ow]]
 ciena_saos_vlan_show.textfsm, .*, ciena_saos, vl[[an]] sh[[ow]]
 
+cisco_asa_show_module.textfsm:cisco_asa_show_module_details.textfsm:cisco_asa_show_module_status.textfsm, .*, cisco_asa, sh[[ow]] modu[[le]]
```
[the remark this answers is not in the document]
Comment (lamiskin) [reply ntc_templates/templates/index]: @jvanderaa I've moved this down now. Apologies, I was basing it off off the cisco ios show module, which had it at the top of its block of commands.

[the remark this answers is not in the document]
Comment (lamiskin) [reply ntc_templates/templates/index]: @jvanderaa The build is now failing. Do you have any suggestions on resolving this?
`Error Message: Current Command len '94' larger then previous '20', for command 'show_module.textfsm:cisco_asa_show_module_details.textfsm:cisco_asa_show_module_status.textfsm'`

**Comment by itdependsnetworks** [comment]
As per readme, follow these steps for ordering. 
image

**Comment by lamiskin** [comment]
> As per readme, follow these steps for ordering. image

@itdependsnetworks I'm stuck with this.
Going by the documentation the new line in the index file should go at the top of the ASA section (due to the use of sub-modules, so the template name is much longer than the rest), but @jvanderaa has rejected this and asked for it to be re-ordered.
Placing it in order with the length of the command as he suggested (from what I assume) causes the tests to fail (because of the length of the template). 

Basing it off what has been done with the cisco_ios show module, it should go at the top of the asa section. 

Unless I've misunderstood, I've got two maintainers telling me different things?

**Comment by jvanderaa** [comment]
@lamiskin you are right. I was wrong on the order. I need to add an item to the documentation to say length order, longest to shortest. My mind went in there and read `length order` and went to shortest to longest.

**Comment by lamiskin** [comment]
@jvanderaa Are you able to merge this?

**Pull request closed without merging**